source/s3_request_messages.c

```diff
@@ -979,12 +953,10 @@ int aws_s3_message_util_copy_headers(
             }
         }
 
-        if (aws_http_message_add_header(dest_message, header)) {
-            return AWS_OP_ERR;
-        }
+        error |= aws_http_message_add_header(dest_message, header);
```

Comment:
This is hard to read. It would be better to just do `AWS_ASSERT(aws_http_message_add_header(dest_message, header))`